aif360/sklearn/detectors/facts/__init__.py

```diff
@@ -0,0 +1,354 @@
+                - `"max-cost-diff-decr-ignore-forall-subgroups-empty"`: ignore \
+                    groups for which we have no available actions whatsoever.
+                - `"max-cost-diff-decr-ignore-exists-subgroup-empty"`: ignore \
+                    groups for which at least one protected subgroup has \
+                    no available actions.
+
+                Defaults to "max-cost-diff-decr".
+
+            top_count (int, optional): the number of subpopulation groups that
+                the algorithm will keep.
+                Defaults to 10.
+
+            filter_sequence (List[str], optional): List of various filters
+                applied on the groups and / or actions. Available filters are:
+
+                - `"remove-contained"`: does not show groups which are subsumed \
+                    by other shown groups. By "subsumed" we mean that the group \
+                    is defined by extra feature values, but those values are \
+                    not changed by any action.
+                - `"remove-below-thr-corr"`: does not show actions which are \
+                    below the given effectiveness threshold. Refer also to the \
+                    documentation of parameter `phi` below.
+                - `"remove-above-thr-cost"`: does not show action that cost more \
+                    than the given cost budget. Refer also to the documentation \
+                    of parameter `c` below.
+                - `"keep-rules-until-thr-corr-reached"`: 
+                - `"remove-fair-rules"`: do not show groups which do not exhibit \
+                    bias.
+                - `"keep-only-min-change"`: for each group shown, show only the \
+                    suggested actions that have minimum cost, ignore the others.
+                
+                Defaults to [].
+            
+            phi (float, optional): effectiveness threshold. Real number in [0, 1].
+                Applicable for "equal-choice-for-recourse" and
+                "equal-cost-of-effectiveness" metrics. For these two metrics, an
+                action is considered to achieve recourse for a subpopulation group
+                if at least `phi` % of the group's individuals achieve recourse.
+                Defaults to 0.5.
+
+            c (float, optional): cost budget. Real number. Applicable for
+                "equal-effectiveness-within-budget" metric. Specifies the maximum
+                cost that can be payed for an action (by the individual, by a
+                central authority etc.)
+                Defaults to 0.5.
+        """
+        if viewpoint == "macro":
+            rules = self.rules_by_if
+        elif viewpoint == "micro":
+            rules = self.rules_with_cumulative
+        else:
+            raise ValueError("viewpoint parameter can be either 'macro' or 'micro'")
+        rules = self.rules_by_if if viewpoint == "macro" else self.rules_with_cumulative
+
+        easy2hard_name_map = {
```

Comment:
Any reason to have a map like this? Why not just use the "easy" names throughout?

Reply:
In short, due to fear of regression. But you are right, it is not needed, it will be removed.

Reply:
Done!